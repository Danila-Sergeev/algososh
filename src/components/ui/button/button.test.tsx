import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Button } from "./button";

describe("<Button />", () => {
    it("рендерит кнопку с текстом", () => {
        const { asFragment } = render(<Button text="Текст кнопки" />);
        expect(screen.getByText("Текст кнопки")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит кнопку без текста", () => {
        const { asFragment } = render(<Button />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("рендерит заблокированную кнопку", () => {
        render(<Button disabled />);
        const button = screen.getByRole("button");
        expect(button).toBeDisabled();
    });

    it("рендерит кнопку с индикацией загрузки", () => {
        const { asFragment } = render(<Button isLoader={true} />);
        expect(screen.getByAltText("Загрузка.")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("корректно вызывает колбек при клике на кнопку", () => {
        const onClickMock = jest.fn();
        render(<Button onClick={onClickMock} />);
        fireEvent.click(screen.getByRole("button"));
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});
