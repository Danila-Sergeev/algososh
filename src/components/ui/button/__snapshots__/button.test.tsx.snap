// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> рендерит кнопку без текста 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`<Button /> рендерит кнопку с индикацией загрузки 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;

exports[`<Button /> рендерит кнопку с текстом 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Текст кнопки
    </p>
  </button>
</DocumentFragment>
`;
